06_puppy/alfredxiao/store_test.go

```diff
@@ -0,0 +1,81 @@
+package main
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type storerSuite struct {
+	suite.Suite
+	store Storer
+}
+
+func (s *storerSuite) TestCreatePuppyHappyCase() {
+	t := s.T()
+	err := s.store.CreatePuppy(Puppy{ID: "1", Colour: "Black"})
+	assert.NoError(s.T(), err, "Happy case puppy creation")
```

Comment:
You've already grabbed a reference to the `testing.T` on the first line of the function.

Reply:
[suite.Suite](https://godoc.org/github.com/stretchr/testify/suite#Suite) also embeds `assert.Assertions` so you can  call `s.NoError(...)` if you would like

Reply:
Nice, didn't know that